src/Tes.Runner/Storage/TerraSasResolutionStrategy.cs

```diff
@@ -0,0 +1,230 @@
+            }
+
+            if (blobSasPermissions.HasFlag(BlobSasPermissions.Delete))
+            {
+                permissions += "d";
+            }
+            return permissions;
+        }
+
+        private void CheckIfAccountIsTerraStorageAccount(string accountName)
+        {
+            if (!IsTerraWorkspaceStorageAccount(accountName))
+            {
+                throw new InvalidOperationException($"The account name does not match the configuration for Terra.");
+            }
+        }
+
+        private bool IsTerraWorkspaceStorageAccount(string value)
+        {
+            var match = Regex.Match(value, LzStorageAccountNamePattern);
+
+            return match.Success;
+        }
+        /// <summary>
+        /// Creates a Terra Blob Info from the container name in the url. The url must be a Terra managed storage URL.
+        /// This method assumes that the container name contains the workspace ID and validates that the storage container is a Terra workspace resource.
+        /// The BlobName property contains the blob name segment without a leading slash.
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>Returns a Terra Blob Info</returns>
+        /// <exception cref="InvalidOperationException">This method will throw if the url is not a valid Terra blob storage url.</exception>
+        private async Task<TerraBlobInfo> GetTerraBlobInfoFromContainerNameAsync(string url)
+        {
+            var blobUriBuilder = ToBlobUriBuilder(url);
+
+            CheckIfAccountIsTerraStorageAccount(blobUriBuilder.AccountName);
+
+            logger.LogInformation($"Getting Workspace ID from the Container Name: {blobUriBuilder.BlobContainerName}");
+
+            var workspaceId = ToWorkspaceId(blobUriBuilder.BlobContainerName);
+
+            logger.LogInformation($"Workspace ID to use: {blobUriBuilder.BlobContainerName}");
+
+            var wsmContainerResourceId = await GetWsmContainerResourceIdAsync(workspaceId, blobUriBuilder.BlobContainerName);
+
+            return new TerraBlobInfo(workspaceId, wsmContainerResourceId, blobUriBuilder.BlobContainerName, blobUriBuilder.BlobName.TrimStart('/'));
+        }
+
+        private BlobUriBuilder ToBlobUriBuilder(string url)
+        {
+            BlobUriBuilder blobUriBuilder;
+            try
+            {
+                blobUriBuilder = new BlobUriBuilder(new Uri(url));
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Failed parse the URL. The URL provided is not a valid Blob URL: {url}");
+                throw;
+            }
+
+            return blobUriBuilder;
+        }
+
+        private Guid ToWorkspaceId(string segmentsContainerName)
+        {
+            try
+            {
+                ArgumentException.ThrowIfNullOrEmpty(segmentsContainerName);
+
+                var guidString = segmentsContainerName.Substring(3); // remove the sc- prefix
```

Comment:
do we need to check if it starts with sc- first?

Reply:
No need. The substring must follow the expected naming convention and be a valid workspace ID. Ther are already checks for this: the substring must be a valid GUID and later there’s a call to WSM to validate that the workspace ID exists, and that the user has access to it.